From f612a5cc0afba6e1af60652283dd47f3f34dcca7 Mon Sep 17 00:00:00 2001
From: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date: Thu, 25 Jun 2015 18:37:58 -0500
Subject: [PATCH 003/146] arch/x86/boot/smbios: Add SPD IDs for Kingston and
 Corsair

---
 src/arch/x86/smbios.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/arch/x86/smbios.c b/src/arch/x86/smbios.c
index a1f05da..74288f9 100644
--- a/src/arch/x86/smbios.c
+++ b/src/arch/x86/smbios.c
@@ -124,10 +124,18 @@ static int smbios_processor_name(char *start)
 void smbios_fill_dimm_manufacturer_from_id(uint16_t mod_id, struct smbios_type17 *t)
 {
 	switch (mod_id) {
+		case 0x9801:
+			t->manufacturer = smbios_add_string(t->eos,
+							    "Kingston");
+			break;
 		case 0x987f:
 			t->manufacturer = smbios_add_string(t->eos,
 							    "Hynix");
 			break;
+		case 0x9e02:
+			t->manufacturer = smbios_add_string(t->eos,
+							    "Corsair");
+			break;
 		case 0xad80:
 			t->manufacturer = smbios_add_string(t->eos,
 							    "Hynix/Hyundai");
-- 
1.7.9.5

