From 10204a5acccb903d407b241501e316506199f60d Mon Sep 17 00:00:00 2001
From: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date: Sat, 5 Sep 2015 19:00:34 -0500
Subject: [PATCH 025/146] include/smbios: Update SMBIOS memory structures to
 version 2.8

---
 src/include/smbios.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/include/smbios.h b/src/include/smbios.h
index b654c23..fdb7bbd 100644
--- a/src/include/smbios.h
+++ b/src/include/smbios.h
@@ -409,9 +409,11 @@ struct smbios_type17 {
 	u8 asset_tag;
 	u8 part_number;
 	u8 attributes;
-	u16 extended_size;
+	u32 extended_size;
 	u16 clock_speed;
-
+	u16 minimum_voltage;
+	u16 maximum_voltage;
+	u16 configured_voltage;
 	char eos[2];
 } __attribute__((packed));
 
-- 
1.7.9.5

