From e25111f4814bf9f7dfd08196673b3fe2aa42f2c4 Mon Sep 17 00:00:00 2001
From: Timothy Pearson <kb9vqf@pearsoncomputing.net>
Date: Thu, 18 Jun 2015 12:37:08 -0500
Subject: [PATCH 063/146] mainboard/asus/kgpe-d16: Set SP5100 subtype

---
 src/mainboard/asus/kgpe-d16/Kconfig |    1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mainboard/asus/kgpe-d16/Kconfig b/src/mainboard/asus/kgpe-d16/Kconfig
index 9c359da..761fc93 100644
--- a/src/mainboard/asus/kgpe-d16/Kconfig
+++ b/src/mainboard/asus/kgpe-d16/Kconfig
@@ -10,6 +10,7 @@ config BOARD_SPECIFIC_OPTIONS # dummy
 	select SOUTHBRIDGE_AMD_SR5650
 	select SOUTHBRIDGE_AMD_SB700
 	select SOUTHBRIDGE_AMD_SB700_DISABLE_ISA_DMA
+	select SOUTHBRIDGE_AMD_SUBTYPE_SP5100
 	select SUPERIO_NUVOTON_NCT5572D
 	select PARALLEL_CPU_INIT
 	select HAVE_ROMSTAGE_CONSOLE_SPINLOCK
-- 
1.7.9.5

