From 5e99bbbf3140b198b2c4b68e646e7042f76806e3 Mon Sep 17 00:00:00 2001
From: Damien Zammit <damien@zamaudio.com>
Date: Sat, 21 May 2016 02:24:19 +1000
Subject: [PATCH] drivers/net/r8168: Add driver for 10ec:8168 to reset the NIC

Change-Id: I32e070b545b4c6369686a7087b7ff838d00764e3
Signed-off-by: Damien Zammit <damien@zamaudio.com>
---

diff --git a/src/drivers/net/Kconfig b/src/drivers/net/Kconfig
new file mode 100644
index 0000000..b4bafd2
--- /dev/null
+++ b/src/drivers/net/Kconfig
@@ -0,0 +1,5 @@
+config REALTEK_8168_RESET
+	bool "Realtek 8168 reset"
+	help
+	  This forces a realtek 10ec:8168 card to reset to ensure power state
+	  is correct at boot.
diff --git a/src/drivers/net/Makefile.inc b/src/drivers/net/Makefile.inc
index 9b3008d..e435d48 100644
--- a/src/drivers/net/Makefile.inc
+++ b/src/drivers/net/Makefile.inc
@@ -1,2 +1,3 @@
 romstage-$(CONFIG_CONSOLE_NE2K) += ne2k.c
 ramstage-$(CONFIG_CONSOLE_NE2K) += ne2k.c
+ramstage-$(CONFIG_REALTEK_8168_RESET) += r8168.c
diff --git a/src/drivers/net/r8168.c b/src/drivers/net/r8168.c
new file mode 100644
index 0000000..be5a7b8
--- /dev/null
+++ b/src/drivers/net/r8168.c
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2015  Damien Zammit <damien@zamaudio.com>
+ *
+ * This driver simply forces the 10ec:8168 device to reset so that it goes
+ * into a proper power state.
+ */
+
+#include <arch/io.h>
+#include <device/device.h>
+#include <device/pci.h>
+#include <device/pci_ids.h>
+#include <device/pci_ops.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define CMD_REG	0x37
+
+static void r8168_init(struct device *dev)
+{
+	u32 cnt = 0;
+
+	/* Get the resource of the NIC mmio */
+	struct resource *nic_res = find_resource(dev, PCI_BASE_ADDRESS_0);
+	u32 nic_mmio = (u32)res2mmio(nic_res, 0, 0);
+
+	/* Reset */
+	outb(0x10, nic_mmio + CMD_REG);
+
+	/* Poll for reset, with timeout */
+	while (cnt < 1000 && (inb(nic_mmio + CMD_REG) & 0x10))
+		cnt++;
+}
+
+static struct device_operations r8168_ops  = {
+       .read_resources   = pci_dev_read_resources,
+       .set_resources    = pci_dev_set_resources,
+       .enable_resources = pci_dev_enable_resources,
+       .init             = r8168_init,
+       .scan_bus         = 0,
+};
+
+static const struct pci_driver r8168_driver __pci_driver = {
+        .ops    = &r8168_ops,
+        .vendor = 0x10ec,
+        .device = 0x8168,
+};
